tensorflow/core/util/ctc/ctc_beam_search_test_double.cc

```diff
@@ -0,0 +1,346 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// This test illustrates how to make use of the CTCBeamSearchDecoder using a
+// custom BeamScorer and BeamState based on a dictionary with a few artificial
+// words.
+#include "tensorflow/core/util/ctc/ctc_beam_search.h"
+
+#include <cmath>
+#include "tensorflow/core/lib/strings/strcat.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace {
+
+typedef std::vector<std::vector<std::vector<double>>> TestData;
+typedef tensorflow::ctc::CTCBeamSearchDecoder<double> CTCBeamSearchDecoder
+typedef tensorflow::ctc::CTCDecoder<double> CTCDecoder;
+
+// The HistoryBeamState is used to keep track of the current candidate and
+// caches the expansion score (needed by the scorer).
+struct HistoryBeamState {
+  double score;
+  std::vector<int> labels;
+};
+
+// DictionaryBeamScorer essentially favors candidates that can still become
```

Comment:
put into a separate PR, please.

Reply:
oh nevermind; just move this into a common test file templated on type float and double.

Reply:
resolved in 51cd5877412f9d67f2ec180c8087ae8d7a059da1